lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -273,6 +273,9 @@ sub _prepare_complex_query_search_args ($self, $args) {
         my $subquery = $schema->resultset('JobGroups')->search({name => $args->{group}})->get_column('id')->as_query;
         push @conds, {'me.group_id' => {-in => $subquery}};
     }
+    if (defined $args->{not_groupid}) {
+      push @conds, {'me.group_id' => {-not_in => $args->{not_groupid}}};
```

Comment:
Just a note, as we do not have groups with id 0, but null: this is turned into
```
WHERE ( me.group_id NOT IN ( 0 ) )
```
which works because in SQL, comparing something with `null` is always false, so any random number would work.